packages/core/assemblyManager.js

```diff
@@ -1,105 +1,26 @@
+import QuickLRU from './util/QuickLRU'
 import { readConfObject } from './configuration'
 
-export default self => ({
-  views: {
-    get assemblyData() {
-      const assemblyData = observable.map({})
-      for (const assemblyConfig of self.assemblies) {
-        const assemblyName = readConfObject(assemblyConfig, 'name')
-        const assemblyInfo = {}
-        if (assemblyConfig.sequence) {
-          assemblyInfo.sequence = assemblyConfig.sequence
-        }
-        const refNameAliasesConf = readConfObject(
-          assemblyConfig,
-          'refNameAliases',
-        )
-        if (refNameAliasesConf) {
-          assemblyInfo.refNameAliases = refNameAliasesConf
-        }
-        const aliases = readConfObject(assemblyConfig, 'aliases')
-        assemblyInfo.aliases = aliases
-        assemblyData.set(assemblyName, assemblyInfo)
-        aliases.forEach((assemblyAlias, idx) => {
-          const newAliases = [
-            ...aliases.slice(0, idx),
-            ...aliases.slice(idx + 1),
-            assemblyName,
-          ]
-          assemblyData.set(assemblyAlias, {
-            ...assemblyInfo,
-            aliases: newAliases,
-          })
-        })
-      }
-      return assemblyData
-    },
-  },
-  actions: {
-    async getCanonicalRefName(refName, assemblyName) {
-      const aliasesToCanonical = await self.getRefNameCanonicalizationMap(
-        assemblyName,
-      )
-      return aliasesToCanonical.get(refName)
-    },
-
-    // returns Map of alias-or-canonical -> canonical
-    async getRefNameCanonicalizationMap(assemblyName, opts = {}) {
-      const refNameAliases = await self.getRefNameAliases(assemblyName, opts)
-      const aliasesToCanonical = new Map()
-      Object.entries(refNameAliases).forEach(([ref, aliases]) => {
-        aliases.forEach(alias => {
-          aliasesToCanonical.set(alias, ref)
-        })
-        aliasesToCanonical.set(ref, ref)
-      })
-      return aliasesToCanonical
-    },
-
-    async getRefNameAliases(assemblyName, opts = {}) {
-      const refNameAliases = {}
-      const assemblyConfig = self.assemblyData.get(assemblyName)
-      if (assemblyConfig && assemblyConfig.refNameAliases) {
-        const adapterConfigId = jsonStableStringify(
-          toJS(assemblyConfig.refNameAliases.adapter),
-        )
-        const adapterRefNameAliases = await self.rpcManager.call(
-          adapterConfigId,
-          'getRefNameAliases',
-          {
-            sessionId: assemblyName,
-            adapterType: assemblyConfig.refNameAliases.adapter.type,
-            adapterConfig: assemblyConfig.refNameAliases.adapter,
-            signal: opts.signal,
-          },
-          { timeout: 1000000 },
-        )
-        adapterRefNameAliases.forEach(alias => {
-          refNameAliases[alias.refName] = alias.aliases
-        })
-      }
-      return refNameAliases
-    },
+export default self => {
+  const cache = new AbortablePromiseCache({
```

Comment:
would be good to rename this to refNameMapCache to make it clear that this cache is only for refname maps